conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanGenerator.java

```diff
@@ -113,125 +108,67 @@ public static JavaFile generateBeanType(
 
         TypeSpec.Builder typeBuilder = TypeSpec.classBuilder(prefixedName.getName())
                 .addModifiers(Modifier.PUBLIC, Modifier.FINAL)
-                .addAnnotations(safety)
-                .addFields(poetFields)
-                .addMethod(createConstructor(fields, poetFields))
-                .addMethods(createGetters(fields, typesMap, options, safetyEvaluator));
+                .addAnnotations(safety);
 
-        if (!poetFields.isEmpty()) {
-            boolean useCachedHashCode = useCachedHashCode(fields);
-            typeBuilder
-                    .addMethod(MethodSpecs.createEquals(objectClass))
-                    .addMethod(MethodSpecs.createEqualTo(objectClass, poetFields, useCachedHashCode));
-            if (useCachedHashCode) {
-                MethodSpecs.addCachedHashCode(typeBuilder, poetFields);
-            } else {
-                typeBuilder.addMethod(MethodSpecs.createHashCode(poetFields));
+        if (!hasFields) {
+            // Need to add JsonSerialize annotation which indicates that the empty bean serializer should be used to
+            // serialize this class. Without this annotation no serializer will be set for this class, thus preventing
+            // serialization.
+            typeBuilder.addAnnotation(JsonSerialize.class).addField(createSingletonField(objectClass));
+            if (!options.strictObjects()) {
+                typeBuilder.addAnnotation(AnnotationSpec.builder(JsonIgnoreProperties.class)
+                        .addMember("ignoreUnknown", "$L", true)
+                        .build());
             }
         }
 
+        typeBuilder.addFields(poetFields);
+        addConstructor(typeBuilder, fields);
```

Comment:
Compared to `typeBuilder.addMethod(createConstructor)`, this approach actually seems worse: we have no guarantee that `addConstructor` doesn't add more than one method.